akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -317,6 +317,22 @@ private[akka] final case class Grouped[T](n: Int) extends PushPullStage[T, immut
 /**
  * INTERNAL API
  */
+private[akka] final case class Limit[T](n: Int) extends PushStage[T, T] {
+  private var left = n
+
+  override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
+    left -= 1
+    if (left >= 0) ctx.push(elem)
+    else {
+      if (ctx.isFinishing) ctx.push(elem)
```

Comment:
This branch is never taken anyway since nobody calls `ctx.absorbTermination()`. The proper fix is to just remove this line.